Support null safety

trustmefelix opened the pull request.

trustmefelix committed:
Upgrade analyzer and glob version

trustmefelix committed:
Upgrade dart_style, source_gen and build package version

trustmefelix committed:
Migrated to null safety version

trustmefelix committed:
Version 2.0.0 - Upgraded dependencies

trustmefelix committed:
Set allocator of DartEmitter as none

trustmefelix committed:
Removed ? character on DartEmitter

trustmefelix committed:
Merge branch 'master' into feature/null_safety

trustmefelix committed:
Bug fix

Fixed issue of Operand of null-aware operation '!' has type 'String' which excludes null

trustmefelix committed:
Removed null safety character on AnnotatedElement

McCrafterIV commented:
Doesn't currently work due to dart_casing and dart_utils not being null safe. See error below:
```
[WARNING] Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:dart_casing
 - package:dart_utils

```

FelixMoMo committed:
Add dependencies of dart casing and utils for support null safety

Add dependencies of dart casing and utils for support null safety - Thanks for McCrafterIV help

FelixMoMo committed:
Removed specific type of generator output and updated app version

LanceQuare commented:
> Doesn't currently work due to dart_casing and dart_utils not being null safe. See error below:
> 
> ```
> [WARNING] Error: Cannot run with sound null safety, because the following dependencies
> don't support null safety:
> 
>  - package:dart_casing
>  - package:dart_utils
> ```

Same error here, even tho dependency overrides to McCrafterIV's git url.

hmayer00 commented:
Looks like @McCrafterIV has pull requests for making dart_casing, dart_utils, and flutter_translate_gen all null-safe (thanks @McCrafterIV!). Is there an expectation that these changes will be merged and published? Thanks.
